### ISSUE TITLE
Remove spaces in column names by default

### ISSUE THREAD

**Issue by nalimilan** [JuliaData/CSV.jl#158]
DataFrames' `readtable` used to remove spaces in column names by default (via a `normalizenames` argument), because they are annoying to type as symbols and because they do not work in query macros like Query or DataFramesMeta. I think CSV should do the same, replacing spaces with `_`.

See https://discourse.julialang.org/t/spaces-in-query-jl/8527.

**Comment by nalimilan** [comment]
Actually it doesn't look like other recent software remove spaces by default, so maybe we should only do it optionally. Copying my comment from Discourse:

R's `read.csv` replaces spaces with `.` by default (dots are the equivalent of underscores in R since they can appear in identifiers). haven doesn't support CSV files, but the accompanying readr package provides `read_csv`, which keeps spaces in column names (without an option to remove them). data.table's `fread` doesn't apply any transformation by default either, but support the `check.names` arguments for that.

So the existing programs adopt a variety of solutions. We could at least support an argument, but whether it should be the default isn't clear.

**Comment by pdeffebach** [comment]
+1 for this. 

I also want to add an option for making variables lowercase, since other software (Stata at least) provides this functionality. 

```
newnames =  map(lowercase, String.(names(df)))
newnames = strip.(newnames)
newnames = Symbol.(newnames)
names!(df, newnames)
```

I will try and put together a pull request for this.

**Comment by SimonAB** [comment]
@pdeffebach have you been able to implement a replacement for `normalizenames` yet?

**Comment by pdeffebach** [comment]
I have some code floating around that overloads a bunch of string operations to work on symbols. But I haven't been focusing on this.

**Comment by quinnj** [comment]
Implemented in #256
